app/controllers/admin/enterprises_controller.rb

```diff
@@ -148,8 +151,15 @@ def find_resource
 
     private
 
-    def load_enterprise_set
-      @enterprise_set = Sets::EnterpriseSet.new(collection) if spree_current_user.admin?
+    def load_enterprise_set_on_index
+      return unless spree_current_user.admin?
+
+      load_enterprise_set_with_params
+    end
+
+    def load_enterprise_set_with_params(params = {})
+      @pagy, @paginated_collection = pagy(@collection)
+      @enterprise_set = Sets::EnterpriseSet.new(@paginated_collection, params)
```

Comment:
I guess that `@paginated_collection` is not used anywhere else in the code. So by making it a local variable, it's clear that you can change the name or do whatever you like without affected other code.

On the other hand, this variable may be useful at some point. And if we publicise `@pagy` then why not the collection? So I don't mind this version either. It looks more consistent.